.github/workflows/ci.yml

```diff
@@ -287,6 +288,44 @@ jobs:
       - run: make test.unit
 
 
+  test-e2e:
+    name: E2E tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: subosito/flutter-action@v2
+        with:
+          flutter-version: $FLUTTER_VER
+          channel: stable
+          cache: true
+
+      - name: Login to container registry
+        run: docker login -u ${{ secrets.E2E_USER }}
+                          -p ${{ secrets.E2E_PASS }}
+                             ${{ secrets.E2E_REGISTRY }}
+
+      - run: make flutter.pub
+
+      - name: Populate .env file to use in Docker compose
+        uses: SpicyPizza/create-envfile@v1.3
+        with:
+          envkey_COMPOSE_BACKEND: ${{ secrets.E2E_BACKEND }}
+          envkey_COMPOSE_COCKROACH: ${{ secrets.E2E_COCKROACH }}
+          envkey_COMPOSE_FRONTEND_IMAGE: nginx
+          envkey_COMPOSE_FRONTEND_TAG: stable-alpine
+          envkey_COMPOSE_MEDEA: ${{ secrets.E2E_MEDEA }}
+          envkey_COMPOSE_PROJECT_NAME: messenger
+          fail_on_empty: true
+
+      - run: mkdir build
+
+      - uses: nanasess/setup-chromedriver@v1
+      - run: |
+          chromedriver --port=4444 --enable-chrome-logs --disable-dev-shm-usage &
+          make test.e2e start-app=yes device=web-server pull=yes no-cache=yes
```

Comment:
@SleepySquash is it possible to do as separate steps?

Reply:
@tyranron, sadly, `flutter driver -d web-server` does not print any logs about the e2e tests progression, just the `All tests passed` or an error occurred. However, `--enable-chrome-logs` flag passed to the `chromedriver` does it, thus I decided to print both logs simultaneously to see the progression in this step. I'm not sure if GitHub Actions allow us to see the logs of previous steps.

Reply:
@SleepySquash such tricks worth mentioning in comments, because are far from being obvious.